PV diverter updates

mattjgalloway opened the pull request:
This fixes #32 and #33.

I've added entities for PV charge limit and PV diverter enabled:
<img width="338" alt="Screenshot 2024-02-14 at 19 36 42" src="https://github.com/tomasmcguinness/homeassistant-mixergy/assets/753010/7a401ad1-dd84-440e-9372-07a042b2c4e9">

In the process, it meant I added `number` and `switch` platforms to the integration. I also pulled out a `MixergyEntityBase` class that puts some of the logic all entities use into one place.

I've tested this on my own tank and it works well.

@tomasmcguinness I wouldn't mind you letting me know if in your settings API call you get the `pv_charge_limit` and `divert_exported_enabled` fields? I'm using absence of those to disable PV related entities. But if they appear in your response, even though you don't have a diverter, then I'll have to think of something different.

tomasmcguinness commented:
I'll check that as soon as I can!

tomasmcguinness commented:
In my settings, I get back this payload:

```
{
    "cleansing_temperature": 60,
    "dsr_enabled": false,
    "automatic_schedule_adjust_target": false,
    "automatic_schedule_heat_source": "indirect",
    "target_temperature_control_enabled": false,
    "frost_protection_enabled": true,
    "distributed_computing_enabled": true,
    "heatpump_topup_mode": false,
    "pv_cut_in_threshold": 0,
    "pv_charge_limit": 100,
    "pv_target_current": -0.2,
    "max_temp": 50,
    "timezone": "Europe/London",
    "automatic_schedule_control": "off",
    "divert_exported_enabled": true,
    "cleansing_schedule": {
        "start": "01:30",
        "days": [
            "fri"
        ]
    }
}
```

mattjgalloway commented:
@tomasmcguinness - sorry, I was away over the weekend so couldn't do any changes on this.

In your API response for the tank, what do you get for this line:
```
"configuration": "{\"tankType\":\"NORMAL\",\"mixergyPvType\":\"RS485_CLAMP_AND_INVERTER\"}"
```

I am wondering if that `mixergyPvType` can tell us if the PV diverter is enabled.

Also by the way - do you have any documentation from Mixergy on the API?

tomasmcguinness commented:
I'll check when I can.

tomasmcguinness commented:
Hey, sorry @mattjgalloway for the delay - 

This is what I get back for my tank

```
"configuration": "{\"tankType\":\"NORMAL\",\"mixergyPvType\":\"NO_INVERTER\"}"
```

tomasmcguinness commented:
Hi @mattjgalloway, there are some merge conflicts in this - I've also replied with the configuration response. Sorry that took so long!

mattjgalloway commented:
Thanks @tomasmcguinness, and no worries about the delay!

I've updated the PR to make it properly only show the PV diverter settings I added if the tank has a diverter. And I resolved the conflicts (were with my own changes to add clamp power).
